libs/datamodel/core/src/transform/ast_to_dml/lift.rs

```diff
@@ -194,24 +193,74 @@ impl<'a> LiftAstToDml<'a> {
-                let args = pg_type_specification
-                    .map(|dir| {
-                        let args = dir
-                            .arguments
-                            .iter()
-                            .map(|arg| ValueValidator::new(&arg.value).as_int().unwrap() as u32)
-                            .collect();
-                        args
-                    })
-                    .unwrap_or(vec![]);
-
-                if let Some(x) = name.and_then(|ts| pg_connector.parse_native_type(&ts, args)) {
-                    let field_type = dml::FieldType::NativeType(scalar_type, x);
+            if supports_native_types {
+                let connector_string = match self.source {
+                    Some(source) => &source.active_provider,
+                    None => "",
+                };
+                let mut connectors: HashMap<&str, Box<dyn Connector>> = HashMap::new();
+                connectors.insert("postgresql", Box::new(SqlDatamodelConnectors::postgres()));
+                connectors.insert("mysql", Box::new(SqlDatamodelConnectors::mysql()));
+                connectors.insert("sqlite", Box::new(SqlDatamodelConnectors::sqlite()));
+                connectors.insert("mssql", Box::new(SqlDatamodelConnectors::mssql()));
+
+                let mut prefix = String::with_capacity(datasource_name.len() + 1);
+                prefix.push_str(datasource_name);
+                prefix.push_str(".");
+
+                let mut type_specifications = ast_field
+                    .directives
+                    .iter()
+                    .filter(|dir| dir.name.name.starts_with(&prefix));
+
+                let type_specification = type_specifications.next();
+
+                if type_specifications.count() > 1 {
+                    return Err(DatamodelError::new_duplicate_directive_error(
+                        &prefix,
+                        type_specification.unwrap().span,
+                    ));
+                }
+
+                let name = type_specification.map(|dir| dir.name.name.trim_start_matches(&prefix));
+
+                // convert arguments to u32 if possible
+                // todo carmen this is really ugly
+                let number_args = type_specification.map(|dir| dir.arguments.clone());
+                let args = if let Some(number) = number_args {
+                    let p = number
+                        .iter()
+                        .map(|arg| ValueValidator::new(&arg.value).as_int())
+                        .collect_vec();
+                    if let Some(error) = p.iter().find(|arg| arg.is_err()) {
+                        return Err(error.clone().err().unwrap());
+                    }
+                    p.iter().map(|arg| *arg.as_ref().unwrap() as u32).collect_vec()
+                } else {
+                    vec![]
+                };
+
+                let connector: &Box<dyn Connector> = connectors.get(connector_string).unwrap();
```

Comment:
You can do this slightly easier. The available `Datasource` already has the connector loaded. Just use the `active_connector` field.